File: async_thread/src/main/java/study/spring/async_thread/service/TestService.java
package study.spring.async_thread.service;

import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

@Service
public class TestService {

  @Async
  public void testMethodWithSleep(int sleepSeeconds) {
    System.out.println("");
    Thread.currentThread().getName()
  }

}

File: async-thread-pool/src/main/kotlin/study/spring/asyncthreadpool/service/TestService.kt
package study.spring.asyncthreadpool.service

import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service

@Service
class TestService {

    @Async
    fun testWithSleep(sleep: Int = 5000, index: Int) {
        val threadName = Thread.currentThread().name
        println("[서브-시작-${index}] ${threadName}")
        Thread.sleep(sleep.toLong())
    }
}